pkg/apis/jenkins.io/v1/types_extensions.go

```diff
@@ -390,6 +390,10 @@ type CommitStatusList struct {
 
 // CommitStatusSpec provides details of a particular commit status
 type CommitStatusSpec struct {
+	Items []CommitStatusDetails `json:"items"  protobuf:"bytes,1,opt,name=items"`
+}
+
+type CommitStatusDetails struct {
 	PipelineActivity ResourceReference           `json:"pipelineActivity"  protobuf:"bytes,1,opt,name=pipelineActivity"`
 	Items            []CommitStatusItem          `json:"Items,omitempty"  protobuf:"bytes,2,opt,name=Items"`
```

Comment:
So this leads to very confusing YAML, such as (note lines 2–3):

```yaml
spec:
  items:
  - Items:
    - description: …
      name: …
      pass: false
    checked: true
    commit:
      gitUrl: https://github.com/….git
      pullRequest: PR-…
      sha: …
    context: …
    pipelineActivity:
      apiVersion: ""
      kind: ""
      name: …
      uid: …
```

I would have expected something more like

```diff
diff --git a/pkg/apis/jenkins.io/v1/types_extensions.go b/pkg/apis/jenkins.io/v1/types_extensions.go
index 553bfdd2..ef30945d 100644
--- a/pkg/apis/jenkins.io/v1/types_extensions.go
+++ b/pkg/apis/jenkins.io/v1/types_extensions.go
@@ -389,12 +389,12 @@ type CommitStatusList struct {
 
 // CommitStatusSpec provides details of a particular commit status
 type CommitStatusSpec struct {
-	Items []CommitStatusDetails `json:"items"  protobuf:"bytes,1,opt,name=items"`
+	Details []CommitStatusDetails `json:"details"  protobuf:"bytes,1,opt,name=details"`
 }
 
 type CommitStatusDetails struct {
 	PipelineActivity ResourceReference           `json:"pipelineActivity"  protobuf:"bytes,1,opt,name=pipelineActivity"`
-	Items            []CommitStatusItem          `json:"Items,omitempty"  protobuf:"bytes,2,opt,name=Items"`
+	Items            []CommitStatusItem          `json:"items,omitempty"  protobuf:"bytes,2,opt,name=items"`
 	Checked          bool                        `json:"checked"  protobuf:"bytes,3,opt,name=checked"`
 	Commit           CommitStatusCommitReference `json:"commit"  protobuf:"bytes,4,opt,name=commit"`
 	Context          string                      `json:"context"  protobuf:"bytes,5,opt,name=context"`
```

Is there a reason the particular format introduced in #2066 and modified here was chosen? The commit message for https://github.com/jenkins-x/jx/pull/2103/commits/bde0f545559233a2b82ccb4b8b4d2a2b6e15ffaf alludes to one but there is no reference.